### DISCUSSION TITLE
generateMetadata doesn't have searchParams - just empty params

### DISCUSSION THREAD

**Discussion by ImLoaD** [vercel/next.js#46849] [Help]
### Summary

According to the documentation, this function should take a searchParams as long as params keys in argument object, but when I try to use it, searchParams is undefined.

```
export async function generateMetadata({ params, searchParams }) {
  // For /products/123, params.id is "123"
  // For /products/123?foo=bar, searchParams.get("foo") is "bar"
  // The return value is the metadata object
  return { title: '...' }
}
```

I even try to capture whole argument:

![image](https://user-images.githubusercontent.com/1041444/223158401-179354e1-1c14-444f-8652-024f71ada939.png)

And no matter what I type in url such as ?foo=bar&some=3 the result is:
![image](https://user-images.githubusercontent.com/1041444/223158586-19554690-c818-4a02-9620-3dfe6afe306b.png)

I used layout.tsx file inside app directory

**Comment by oBusk** [comment]
@ImLoaD what version of Next were you using when you had this issue. 

The `generateMetadata()` method would be called without params as of `v13.1.6`, but should work as expected as of `v13.2.0`.

Are you also using any segment configuration such as [`export const dynamic = 'force-static'`](https://beta.nextjs.org/docs/api-reference/segment-config#dynamic) or [`generateStaticParams()`](https://beta.nextjs.org/docs/api-reference/segment-config#generatestaticparams)? Forcing it to be static will cause `searchParams` to be empty.

**Reply by AndresSalomon1990** [reply]
I'm having a similar problem related to static export. Is there a way to make the generateMetadata work for these static exports?

**Comment by Jens-n11** [comment]
I tested it with newest next version `13.4.19` and [`export const dynamic = 'force-dynamic'`](https://beta.nextjs.org/docs/api-reference/segment-config#dynamic) inside the `layout.tsx` and the problem still exists as described.
-> So `params` are empty `{}` and `searchParams` are `undefined`

But the same code works in a `page.tsx`. Probably this is a bug?

**Reply by Jens-n11** [reply]
Did a little bit more of digging into it. Seems like this is expected behavior:
https://nextjs.org/docs/app/api-reference/file-conventions/layout#layouts-do-not-receive-searchparams

**Reply by minpeter** [reply]
Thank you, it helped me a lot.
